rerun_cpp/tests/component_batch.cpp

```diff
@@ -117,12 +118,8 @@ namespace rerun {
     template <>
     struct ComponentBatchAdapter<components::Position2D, MyVec2Container> {
         ComponentBatch<components::Position2D> operator()(const MyVec2Container& container) {
-            // Sanity check that this is binary compatible.
-            static_assert(sizeof(components::Position2D) == sizeof(float) * 2);
-            static_assert(alignof(components::Position2D) <= sizeof(float));
```

Comment:
Isn't it good to have these sanity checks? 😬

Reply:
…especially since you've changed the `reinterpret_cast<const components::Position2D*>` into a `reinterpret_cast<const void*>`. Sure, this is only a test, but still

Reply:
ah true! At first I thought they all move to ComponentBatch and then on this test I noticed that it wasn't possible here but then I didn't bring back the sanity (check)!